mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp

```diff
@@ -1007,6 +1007,12 @@ void NVVM::WgmmaMmaAsyncOp::getAsmValues(
   }
 }
 
+LogicalResult NVVM::SetMaxRegisterOp::verify() {
+  if (getCount() % 8)
+    return emitOpError("new register size must be multiple of 8");
+  return success();
```

Comment:
I think we can add the range check also:
if (getCount() < 24 || getCount() > 256)
  return emitOpError("...out-of-range..");

Reply:
good catch, I've added this one as well.